### PR TITLE
fix: grid list styles

### PR DESCRIPTION
**Pull request by Betrozov**
## Related Issue
Closes https://github.com/SAP/fundamental-ngx/issues/4560

## Description
Fixed label focus style and added overflow for item

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [X] All values are in `rem`
- [X] Text elements follow the truncation rules
- [X] hover state of the element follow design spec
- [X] focus state of the element follow design spec
- [X] active state of the element follow design spec
- [X] selected state of the element follow design spec
- [X] selected hover state of the element follow design spec
- [X] pressed state of the element follow design spec
- [X] Responsiveness rules - the component has modifier classes for all breakpoints
- [X] Includes Compact/Cosy/Tablet design
- [X] RTL support
2. The code follows fundamental-styles code standards and style
- [X] only one top level `fd-*` class is used in the file
- [X] BEM naming convention is used
- [X] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [X] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [X] `fd-reset()` mixin is applied to all elements
- [X] Variables are used, if some value is used more than twice.
- [X] Checked if current components can be reused, instead of having new code.
3. Testing
- [X] tested Storybook examples with "CSS Resources" `normalize` option 
- [X] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [X] Verified all styles in IE11
- [X] Updated tests
- [X] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [X] Storybook documentation has been created/updated
- [X] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

**Review by IlyaSurmay — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Comment by InnaAtanasova** [comment]
You need a commit message containing `[ci visual]` to trigger Chromatic

File: src/list-grid.scss [hunk 1]
```diff
@@ -47,10 +48,13 @@ $block: #{$fd-namespace}-grid-list;
       @include fd-focus() {
         + .#{$block}__radio-label {
           outline-offset: 0;
+          outline: var(--sapContent_FocusWidth) var(--sapContent_FocusStyle) var(--sapContent_FocusColor);
         }
       }
 
       + .#{$block}__radio-label {
+        @include fd-fiori-focus(0);
+
```
Comment (JKMarkowski) [inline src/list-grid.scss:57]:
```suggestion
      + .#{$block}__radio-label {
        @include fd-fiori-focus(0);
```

**Review by JKMarkowski — COMMENTED** [review]
Hello @Betrozov I think there is a lot of unnecessary code, check my suggestion - `input[type=radio]` element is never visible

**Review by JKMarkowski — CHANGES_REQUESTED** [review]
Check my comment above

File: src/list-grid.scss [hunk 2]
```diff
@@ -47,10 +48,13 @@ $block: #{$fd-namespace}-grid-list;
       @include fd-focus() {
         + .#{$block}__radio-label {
           outline-offset: 0;
+          outline: var(--sapContent_FocusWidth) var(--sapContent_FocusStyle) var(--sapContent_FocusColor);
         }
       }
 
       + .#{$block}__radio-label {
+        @include fd-fiori-focus(0);
```
Comment (JKMarkowski) [inline src/list-grid.scss:56]: So this can be removed

**Review by JKMarkowski — APPROVED** [review]

**Review by Seamoo13 — APPROVED** [review]

**Pull request merged**